code/game/atoms_movable.dm

```diff
@@ -960,14 +964,14 @@
 /// Only moves the object if it's under no gravity
 /atom/movable/proc/newtonian_move(direction)
 	if(!isturf(loc) || Process_Spacemove(0))
-		inertia_dir = 0
 		return FALSE
 
-	inertia_dir = direction
-	if(!direction)
+	if(SEND_SIGNAL(src, COMSIG_MOVABLE_NEWTONIAN_MOVE, direction) & COMPONENT_MOVABLE_NEWTONIAN_BLOCK)
 		return TRUE
-	inertia_last_loc = loc
-	SSspacedrift.processing[src] = src
+
+	set_glide_size(MOVEMENT_ADJUSTED_GLIDE_SIZE(inertia_move_delay, SSspacedrift.visual_delay))
+	AddComponent(/datum/component/drift, direction)
```

Comment:
fear, have you profiled this?

Reply:
I did some testruns and things didn't seem too off.
It's gonna be more expensive because we're running every tick rather then every 10, but it makes things look so much nicer. Don't think it's that bad.

I don't think the actual component add is too big a deal. It's more expensive, but it's a one time thing, and not too common of a one time thing.

Will get you better numbers on both the tick to tick operation and component stuff

Reply:
Tested by dumping 1200 candy cigs out an airlock in cargo
Component add is cheap it seems
![image](https://user-images.githubusercontent.com/58055496/148873022-c5e7b8af-180f-4e0c-ab02-47be894dc031.png)
Self cost is high, but not horrid. Total's skewed cause I dumped em all out at once, so there's a lot of like (Increases with tile contents count) suffering
![image](https://user-images.githubusercontent.com/58055496/148872972-8ccabb08-d588-4184-96b3-23884c4bdf94.png)

Reply:
Here was a test done by dumping 1200 items out an airlock with gas pressure
Done on both old and new code, new split up a bit which I expected would drop Move()'s total cost, but it didn't.

Reply:
[old.csv](https://github.com/tgstation/tgstation/files/7845698/old.csv)
[new.csv](https://github.com/tgstation/tgstation/files/7845699/new.csv)